src/h3lib/include/mathExtensions.h

```diff
@@ -28,12 +29,22 @@
 #define MAX(a, b) (((a) > (b)) ? (a) : (b))
 
 /** Evaluates to true if a + b would overflow for int32 */
-#define ADD_INT32S_OVERFLOWS(a, b) \
-    ((a) > 0 ? (INT32_MAX - (a) < (b)) : (INT32_MIN - (a) > (b)))
+static inline bool ADD_INT32S_OVERFLOWS(int32_t a, int32_t b) {
+    if (a > 0) {
+        return INT32_MAX - a < b;
+    } else {
+        return INT32_MIN - a > b;
+    }
+}
```

Comment:
What is the purpose of these two changes? Do they actually affect the compiler warnings in some way?

Reply:
It's for suppressing the following warnings (that are listed in the description):

src/h3lib/lib/coodijk.c:

```text
(236,13): warning C4307: '+': signed integral constant overflow
```

src/apps/testapps/testMathExtensionsInternal.c:

```text
(81,9): warning C4307: '-': signed integral constant overflow
(82,9): warning C4307: '-': signed integral constant overflow
(83,9): warning C4307: '-': signed integral constant overflow
(84,9): warning C4307: '-': signed integral constant overflow
(85,9): warning C4307: '-': signed integral constant overflow
(86,9): warning C4307: '-': signed integral constant overflow
(87,9): warning C4307: '-': signed integral constant overflow
(88,9): warning C4307: '-': signed integral constant overflow
(89,9): warning C4307: '-': signed integral constant overflow
(90,9): warning C4307: '-': signed integral constant overflow
(91,9): warning C4307: '-': signed integral constant overflow
(92,9): warning C4307: '-': signed integral constant overflow
(93,9): warning C4307: '-': signed integral constant overflow
(94,9): warning C4307: '-': signed integral constant overflow
(95,9): warning C4307: '-': signed integral constant overflow
(96,9): warning C4307: '-': signed integral constant overflow
(97,9): warning C4307: '-': signed integral constant overflow
(98,9): warning C4307: '-': signed integral constant overflow
(99,9): warning C4307: '-': signed integral constant overflow
(100,9): warning C4307: '-': signed integral constant overflow
(101,9): warning C4307: '-': signed integral constant overflow
(102,9): warning C4307: '-': signed integral constant overflow
(103,9): warning C4307: '-': signed integral constant overflow
(104,9): warning C4307: '-': signed integral constant overflow
(105,9): warning C4307: '-': signed integral constant overflow
(106,9): warning C4307: '-': signed integral constant overflow
```

It seems that MSVC can't process `... ? ... : ...` correctly for this warning yet.